src/synapse/api.py

```diff
@@ -212,6 +241,6 @@ def get_version() -> str:
     except requests.exceptions.HTTPError as exc:
         logger.exception("HTTP error from %s: %r", VERSION_URL, exc)
         raise NetworkError(f"HTTP error from {VERSION_URL}.") from exc
-    except GetVersionError as exc:
+    except (GetVersionError, requests.exceptions.JSONDecodeError) as exc:
```

Comment:
Ah, probably `requests.exceptions.JSONDecodeError` needs to be handled whenever we try to convert a response to JSON

Reply:
Changed that too.